shell/src/main/java/alluxio/cli/docgen/MetricsDocGenerator.java

```diff
@@ -110,6 +148,17 @@ public static void generate() throws IOException {
             key, StringEscapeUtils.escapeHtml4(metricKey.getDescription().replace("'", "''"))));
       }
     }
+    if (validate) {
+      boolean hasDiffCSV = false;
+      boolean hasDiffYML = false;
+      for (String category : CATEGORIES) {
+        hasDiffCSV = compareFiles(csvFolder, CSV_SUFFIX, category, hasDiffCSV);
```

Comment:
rather than passing the boolean variable into the function, i find it more intuitive to do the boolean logic outside the function:
`hasDiffCSV = compareFiles(csvFolder, CSV_SUFFIX, category, hasDiffCSV) || hasDiffCSV;`
note that the order is important; i want to avoid short circuting because i always want to run compareFiles to print the file names of nonmatching files if hasDiffCSV has already been set to true

also if going along with this pattern, there is no longer a need for 2 separate boolean variables